### PR TITLE
Public API changes to QueryMethodProvider

### PR DESCRIPTION
**Pull request by roji**

### CODE REVIEW

File: src/EFCore/Query/QueryableMethods.cs [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+        public static bool IsSumWithSelector(MethodInfo methodInfo)
+            => methodInfo.IsGenericMethod
+               && SumWithSelectorMethods.Values.Contains(methodInfo.GetGenericMethodDefinition());
+        public static bool IsAverageWithoutSelector(MethodInfo methodInfo)
+            => AverageWithoutSelectorMethods.Values.Contains(methodInfo);
+        public static bool IsAverageWithSelector(MethodInfo methodInfo)
+            => methodInfo.IsGenericMethod
+               && AverageWithSelectorMethods.Values.Contains(methodInfo.GetGenericMethodDefinition());
+
+        public static IReadOnlyDictionary<Type, MethodInfo> SumWithoutSelectorMethods { get; }
```
Comment (smitpatel) [inline src/EFCore/Query/QueryableMethods.cs]: Didn't we decide to expose method to get method info for particular type rather than whole dictionary?

Comment (ajcvickers) [reply src/EFCore/Query/QueryableMethods.cs]: Yes--I missed this in my review. @roji?

Comment (roji) [reply src/EFCore/Query/QueryableMethods.cs]: Oh, sorry, I thought we said changing to read-only is enough - there could conceivable be some value in allowing users to iterate over the keys/values. But I can of course change this if you guys prefer a get method (GetSumWithoutSelector?)

Comment (ajcvickers) [reply src/EFCore/Query/QueryableMethods.cs]: I don't feel strongly about it.

Comment (smitpatel) [reply src/EFCore/Query/QueryableMethods.cs]: Convert to method which takes type. Current usages does not require iterating over values. We could add dictionary later if situation requires it.

Comment (roji) [reply src/EFCore/Query/QueryableMethods.cs]: #17168

**Pull request merged**